src/transport/websocket/mod.rs

```diff
@@ -385,6 +438,24 @@ impl Transport for WebSocketTransport {
             .map_or(Err(Error::ConnectionDoesntExist(connection_id)), |_| Ok(()))
     }
 
+    fn accept_pending(&mut self, connection_id: ConnectionId) -> crate::Result<()> {
+        let pending = self
+            .pending_inbound_connections
+            .remove(&connection_id)
+            .ok_or(Error::ConnectionDoesntExist(connection_id))?;
+
+        self.on_inbound_connection(connection_id, pending.connection, pending.address);
+
+        Ok(())
+    }
+
+    fn reject_pending(&mut self, connection_id: ConnectionId) -> crate::Result<()> {
+        self.canceled.insert(connection_id);
```

Comment:
```suggestion
```

I don't think we should insert the connection ID into `canceled`, as this is not an outbound connection established without negotiation (see https://github.com/paritytech/litep2p/issues/197 for a detailed explanation).

Reply:
Yep that makes sense! Nice catch 🙏